### ISSUE TITLE
Stalling on period changes in vod manifest

### ISSUE THREAD

**Issue by phildupuis** [google/shaka-player#1889]
Hi !

I'm presently having some troubles with period changes. The problem is that the player gets stalled (video and audio) on period changes. We are having gaps between those periods and perhaps this is the reason why we get stalled. For the vod example below we are experiencing those gaps on the shaka player demo app. 

1. `Jumping forward 0.046521 seconds because of gap before start time of 0.066733`
2. `Jumping forward 0.13628500000000088 seconds because of gap starting at 30.396966 and ending at 30.496728`

```
<?xml version="1.0" encoding="UTF-8"?>
<MPD xmlns="urn:mpeg:dash:schema:mpd:2011" xmlns:cenc="urn:mpeg:cenc:2013" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" maxSegmentDuration="PT2.970S" mediaPresentationDuration="PT47M13.256496S" minBufferTime="PT0S" profiles="urn:mpeg:dash:profile:isoff-live:2011" type="static">
  <Period id="430da13b-c735-4447-95df-45a57ee153e3" start="PT0S">
    <BaseURL>https://base-manifest-url</BaseURL>
      <AdaptationSet id="1" group="1" bitstreamSwitching="true" segmentAlignment="true" contentType="video" mimeType="video/mp4" maxWidth="1280" maxHeight="720" par="16:9" maxFrameRate="30000/1001" startWithSAP="1">
         <SegmentTemplate timescale="10000000" media="S!dxIDVP7...8BFgZ4IOMBFQEWAa8_/QualityLevels($Bandwidth$)/Fragments(video=$Time$)" initialization="S!dxIDVP7...8BFgZ4IOMBFQEWAa8_/QualityLevels($Bandwidth$)/Fragments(video=Init)">
            <SegmentTimeline>
               <S d="20020000" />
               <S d="20019889" />
               <S d="20020000" r="1" />
               <S d="20019889" />
               <S d="20020000" />
               <S d="20019889" />
               <S d="20020000" r="1" />
               <S d="20019889" />
               <S d="20020000" />
               <S d="20019889" />
               <S d="20020000" r="1" />
               <S d="24020555" />
            </SegmentTimeline>
         </SegmentTemplate>
         <Representation id="dzGqZQmf" bandwidth="615850" codecs="avc1.4d401f" width="640" height="360" frameRate="30000/1001" sar="1:1" />
         <Representation id="dzF_gBKf" bandwidth="1212543" codecs="avc1.4d401f" width="720" height="480" frameRate="30000/1001" sar="1:1" />
         <Representation id="dzGFIx2f" bandwidth="1909637" codecs="avc1.4d401f" width="960" height="540" frameRate="30000/1001" sar="1:1" />
         <Representation id="dzGUozif" bandwidth="3711892" codecs="avc1.4d4029" width="1280" height="720" frameRate="30000/1001" sar="1:1" />
         <Representation id="dzHaFlSf" bandwidth="5510874" codecs="avc1.4d4029" width="1280" height="720" frameRate="30000/1001" sar="1:1" />
      </AdaptationSet>
      <AdaptationSet id="2" group="2" bitstreamSwitching="true" segmentAlignment="true" contentType="audio" mimeType="audio/mp4" lang="und">
         <AudioChannelConfiguration schemeIdUri="urn:mpeg:dash:23003:3:audio_channel_configuration:2011" value="2" />
         <SegmentTemplate timescale="48000" media="S!dxIDVP7...8BFgZ4IOMBFQEWAa8_/QualityLevels($Bandwidth$)/Fragments(audio_483_und=$Time$)" initialization="S!dxIDVP7...8BFgZ4IOMBFQEWAa8_/QualityLevels($Bandwidth$)/Fragments(audio_483_und=Init)">
            <SegmentTimeline>
               <S t="446" d="96255" />
               <S d="96257" />
               <S d="96256" />
               <S d="95232" />
               <S d="96255" />
               <S d="96256" />
               <S d="96257" />
               <S d="96256" r="1" />
               <S d="95231" />
               <S d="96256" />
               <S d="96257" />
               <S d="96256" r="1" />
               <S d="114688" />
            </SegmentTimeline>
         </SegmentTemplate>
         <Representation id="dyHjAZ8." bandwidth="192000" codecs="mp4a.40.2" audioSamplingRate="48000" />
      </AdaptationSet>
      <AdaptationSet id="3" group="4" bitstreamSwitching="true" segmentAlignment="true" contentType="text" mimeType="application/mp4" lang="und">
         <Role schemeIdUri="urn:mpeg:dash:role:2011" value="caption" />
         <SegmentTemplate timescale="10000000" media="S!dxIDVP7...8BFgZ4IOMBFQEWAa8_/QualityLevels($Bandwidth$)/Fragments(caption_und_0=$Time$)" initialization="S!dxIDVP7...8BFgZ4IOMBFQEWAa8_/QualityLevels($Bandwidth$)/Fragments(caption_und_0=Init)">
            <SegmentTimeline>
               <S d="20020000" />
               <S d="20019889" />
               <S d="20020000" r="1" />
               <S d="20019889" />
               <S d="20020000" />
               <S d="20019889" />
               <S d="20020000" r="1" />
               <S d="20019889" />
               <S d="20020000" />
               <S d="20019889" />
               <S d="20020000" r="1" />
               <S d="24020555" />
            </SegmentTimeline>
         </SegmentTemplate>
         <Representation id="dxHInw.." bandwidth="200" codecs="stpp" />
      </AdaptationSet>
  </Period>
  <Period id="ce594563-a630-420f-a6a2-0e3ad2c6e970" start="PT30.429995S">
    <BaseURL>https://base-manifest-url</BaseURL>
    <AdaptationSet id="1" group="1" bitstreamSwitching="true" segmentAlignment="true" contentType="video" mimeType="video/mp4" maxWidth="1280" maxHeight="720" par="16:9" maxFrameRate="30000/1001" startWithSAP="1">
         <ContentProtection schemeIdUri="urn:uuid:edef8ba9-79d6-4ace-a3c8-27dcd51d21ed" value="Widevine" />
         <ContentProtection schemeIdUri="urn:mpeg:dash:mp4protection:2011" value="cenc" />
         <SegmentTemplate timescale="10000000" media="S!dxIDVP7...8BFgZnDU9QUEUtV0lERVZJTkV4IOMBFQEWAa8_/QualityLevels($Bandwidth$)/Fragments(video=$Time$)" initialization="S!dxIDVP7...8BFgZnDU9QUEUtV0lERVZJTkV4IOMBFQEWAa8_/QualityLevels($Bandwidth$)/Fragments(video=Init)">
            <SegmentTimeline>
               <S d="20020000" />
               <S d="20019889" />
               <S d="20020000" r="1" />
               <S d="20019889" />
               <S d=".....
```

There is probably something wrong with this manifest, but we can't figure out what is the problem. We are using the exact same manifest with other Dash player and we do not seem to have the problem. Do you have an idea what is wrong with this manifest and what changes could be made to fix the problem ? Or is this a known issue ? Some existing issues refer to period changes stalling, but none of them could help me.

Thank you :)

**Comment by maxi297** [comment]
Hey there! I'm working with @phildupuis on this. We have seen similar behaviour in the past for the live content where we actually had a gap in the manifest (see: https://github.com/google/shaka-player/issues/1864). The impact is the same: on the browser the player can recover but on chromecast the player stalls. However right now, it seems to us that the manifest does not contain such gaps and that the player shouldn't even attempt to jump a gap.

If some debugging is needed on your side, we are willing to provide a streaming and license URL for you to test in private.

Thanks a lot guys!

**Comment by TheModMaker** [comment]
Please file this as a bug.  That template includes important questions for things we need to help.  Including the platform you are running on and requesting a playable manifest URL.  Debugging with just the manifest text is rarely enough.

We have noticed that on some platform, mainly Chromecast, it takes a while to recover after a seek or stalling.  You can fiddle with the last few properties of the [streaming configuration](https://nightly-dot-shaka-player-demo.appspot.com/docs/api/shaka.extern.html#.StreamingConfiguration) to help.  For example, setting `streaming.stallEnabled` to `false` may help.  Also what logs do you see with more verbose logging (use a debug/uncompiled build)?

**Comment by mimse** [comment]
@TheModMaker is there any way to pass these configurations from CAF SDK down to shaka-player?

**Comment by mimse** [comment]
After some debugging, I found out it is possible to set

`PlayerConfig: { 
 shakaConfig: {
  streaming: {
     stallEnabled: false
   } 
 }
}`

But the current version of shaka-player on chromecast is 2.5.0-beta3 which does not contain the stall detection logic.
So I cannot test it.

**Comment by TheModMaker** [comment]
I'm not that familiar with the CAF API.  One option would be to try with a custom receiver app based on just Shaka Player.  You can use Shaka Player directly without CAF and see if that helps your content, that would allow you to test with the latest `master`.  You can take a look at our [cast receiver](https://github.com/google/shaka-player/tree/master/demo/cast_receiver) for a starting point.

Those configurations will appear in the v2.5 release, which we hope to make within the next few weeks.  Hopefully CAF will update with that version soon after.

**Comment by maxi297** [comment]
We were trying to repro when you asked for more information and were unable to. After a little bit of investigation, we saw that Google did update the version of Shaka in the Cast framework to v2.5.0-beta3 which seem to have solved our issue. Therefore, we will then consider this as solved.

P.S.: For someone with the same issue: as of 2019-04-26 Google did rollback the shaka-player version to 2.4.4 which causes us to issue again.

**Comment by TheModMaker** [comment]
The CAF team needed to rollback the v2.5 usage since it caused issues with some DRM content.  We hope to make the final v2.5 release within a week and CAF will hopefully update to that soon.  Since you said it was fixed with v2.5.0-beta3, I'll close this issue.  Feel free to reopen or post a new issue if your problem persists after CAF updates to v2.5.
